### PR TITLE
Add dynamic features

### PR DESCRIPTION
**Pull request by DrNochi**
- In-memory db (Should probably be changed to persistent db later, bu in-memory ensures a fixed state after a restart for testing purposes at the moment)
- Basic flashcard adding (~~no categories yet~~)
- Basic display of categories (~~only name as of yet~~)

### CODE REVIEW

**Comment by Maaxxs** [comment]
How to handle empty database values?
If no value provided as [here](https://github.com/manuel-tiedtke/M2N-Webanwendungen-2/blob/babe8eec607ff64738505c2a6d5a2e891ae98fc8/src/main/java/org/m2n/webapplications2/database/helpers/DbInit.java#L55), then it will display `undefined` in the Dashboard.

[](https://imgur.com/hatdO4w.png)

So
1. Don't allow empty values in the database in the first place?
2. Catch that in [here](https://github.com/manuel-tiedtke/M2N-Webanwendungen-2/blob/1ac37f154f39719348842c553cd35faf1f684904/src/main/java/org/m2n/webapplications2/database/helpers/DbCategory.java#L42)

File: src/main/java/org/m2n/webapplications2/database/helpers/DbInit.java [hunk 1]
```diff
@@ -56,10 +56,10 @@ public static void insertTestData() throws DatabaseException {
...
             statement.addBatch("INSERT INTO category (name, tagline, description)\n" +
                 "VALUES\n" +
-                "('Python', 'Dunno', 'Python is awesome'),\n" +
-                "('Food', 'Dunno', 'Eat! Or you''ll starve!'),\n" +
-                "('Health', 'Dunno', 'See category Food to stay healthy'),\n" +
-                "('Hardware', 'Dunno', 'Hardware Hacking Topics');\n");
+                "('Python', 'Programming Language', 'Python is awesome'),\n" +
+                "('Food', 'Maultaschen', 'Eat! Or you''ll starve!'),\n" +
+                "('Health', 'Sports', 'See category Food to stay healthy'),\n" +
+                "('Hardware', 'Security', 'Hardware Hacking Topics');\n");
```
Comment (DrNochi) [inline src/main/java/org/m2n/webapplications2/database/helpers/DbInit.java:62]: Please also change the corresponding `.sql` file (should be `/src/main/sql/insert-test.sql` or something like that)

Comment (Maaxxs) [reply src/main/java/org/m2n/webapplications2/database/helpers/DbInit.java:62]: Done. 
429d830

**Comment by DrNochi** [comment]
> How to handle empty database values?
> 1. Don't allow empty values in the database in the first place?
> 2. Catch that in [here](https://github.com/manuel-tiedtke/M2N-Webanwendungen-2/blob/1ac37f154f39719348842c553cd35faf1f684904/src/main/java/org/m2n/webapplications2/database/helpers/DbCategory.java#L42)

1. It should be allowed to not have a description or taglines
2. This "bug" should probably be handles in the `dashboard.js` file, as it is there that the value undefined arises. E.g. if the `description` is `null` in Java Jersey probably does not serialize this attribute. If JavaScript now tries to access the `description` in the deserialized version, it reads undefined.

Possible solution like:
`template.querySelector('h6').innerText = category.tagline ? category.tagline : ''`

File: src/main/java/org/m2n/webapplications2/database/helpers/DbInit.java (hunk 1 above)
Comment (DrNochi) [reply src/main/java/org/m2n/webapplications2/database/helpers/DbInit.java:62]: Thanks 👍

**Comment by Maaxxs** [comment]
> > How to handle empty database values?
> > 
> > 1. Don't allow empty values in the database in the first place?
> > 2. Catch that in [here](https://github.com/manuel-tiedtke/M2N-Webanwendungen-2/blob/1ac37f154f39719348842c553cd35faf1f684904/src/main/java/org/m2n/webapplications2/database/helpers/DbCategory.java#L42)
> 
>     1. It should be allowed to not have a description or taglines
> 
>     2. This "bug" should probably be handles in the `dashboard.js` file, as it is there that the value undefined arises. E.g. if the `description` is `null` in Java Jersey probably does not serialize this attribute. If JavaScript now tries to access the `description` in the deserialized version, it reads undefined.
> 
> 
> Possible solution like:
> `template.querySelector('h6').innerText = category.tagline ? category.tagline : ''`

I like that.

**Pull request merged**